playground/signalr/SignalR.AppHost/Program.cs

```diff
@@ -1,9 +1,41 @@
+using Aspire.Hosting.Azure;
+using Azure.Provisioning;
+using Azure.Provisioning.SignalR;
+
 var builder = DistributedApplication.CreateBuilder(args);
 
-var signalr = builder.AddAzureSignalR("signalr1");
+// Configure Azure SignalR in default mode
+var defaultSignalr = builder.AddAzureSignalR("signalrDefault");
 
 builder.AddProject<Projects.SignalRWeb>("webfrontend")
     .WithExternalHttpEndpoints()
-    .WithReference(signalr);
+    .WithReference(defaultSignalr);
+
+// Configure Azure SignalR in serverless mode
+var serverlessSignalr = builder
+    .AddAzureSignalR("signalrServerless")
+    .ConfigureInfrastructure(infra =>
+    {
+        var resource = infra.GetProvisionableResources().OfType<SignalRService>().First(s => s.BicepIdentifier == infra.AspireResource.GetBicepIdentifier());
+        resource.Features.Add(new SignalRFeature()
+        {
+            Flag = SignalRFeatureFlag.ServiceMode,
+            Value = "Serverless"
+        });
+        // Add required role assignment to invoke REST API for serverless mode
+        var principalTypeParameter = infra.GetProvisionableResources().OfType<ProvisioningParameter>().First(o => o.BicepIdentifier == AzureBicepResource.KnownParameters.PrincipalType);
+        var principalIdTypeParameter = infra.GetProvisionableResources().OfType<ProvisioningParameter>().First(o => o.BicepIdentifier == AzureBicepResource.KnownParameters.PrincipalId);
+        infra.Add(resource.CreateRoleAssignment(SignalRBuiltInRole.SignalRRestApiOwner, principalTypeParameter, principalIdTypeParameter));
```

Comment:
@vicancy can we make this a first class API on the AzureSignalR hosting integration?

Reply:
@davidfowl I can add a parameter (e.g serviceMode) to current `AddAzureSignalR` method, and provision the service with the corresponding role assignments based on this parameter

Reply:
@davidfowl @vicancy could I include this improvement in another PR and try to ship this one first for next release? This is a change to the current public API and since we also have holidays next week so it's quite risky to include this in the current PR

Reply:
let me do that.

Reply:
updated PR with above API change